ginivision/src/main/java/net/gini/android/vision/review/multipage/MultiPageReviewFragment.java

```diff
@@ -0,0 +1,493 @@
+        if (activity == null) {
+            return;
+        }
+
+        final LinearLayoutManager layoutManager = new LinearLayoutManager(activity,
+                LinearLayoutManager.HORIZONTAL, false);
+        mThumbnailsRecycler.setLayoutManager(layoutManager);
+
+        mThumbnailChangeListener = new ThumbnailChangeListener() {
+            @Override
+            public void onThumbnailMoved() {
+                final PagerAdapter adapter = mPreviewsPager.getAdapter();
+                if (adapter != null) {
+                    adapter.notifyDataSetChanged();
+                }
+            }
+
+            @Override
+            public void onThumbnailSelected(final int position) {
+                mPreviewsPager.setCurrentItem(position);
+            }
+        };
+
+        mThumbnailsAdapter = new ThumbnailsAdapter(activity, mMultiPageDocument,
+                mThumbnailChangeListener);
+        mThumbnailsRecycler.setAdapter(mThumbnailsAdapter);
+
+        mThumbnailsScroller = new LinearSmoothScroller(activity);
+
+        final ItemTouchHelper.Callback callback =
+                new ThumbnailsTouchHelperCallback(mThumbnailsAdapter);
+        final ItemTouchHelper touchHelper = new ItemTouchHelper(callback);
+        touchHelper.attachToRecyclerView(mThumbnailsRecycler);
+        mThumbnailsAdapter.setItemTouchHelper(touchHelper);
+    }
+
+    private void bindViews(final View view) {
+        mRootView = view.findViewById(R.id.root_view);
+        mButtonNext = view.findViewById(R.id.gv_button_next);
+        mPreviewsPager = view.findViewById(R.id.gv_view_pager);
+        mPageIndicator = view.findViewById(R.id.gv_page_indicator);
+        mThumbnailsRecycler = view.findViewById(R.id.gv_thumbnails_panel);
+        mReorderButton = view.findViewById(R.id.gv_button_reorder);
+        mRotateButton = view.findViewById(R.id.gv_button_rotate);
+        mDeleteButton = view.findViewById(R.id.gv_button_delete);
+    }
+
+    private void setInputHandlers() {
+        mButtonNext.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(final View v) {
+                onNextButtonClicked();
+            }
+        });
+        mReorderButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(final View v) {
+                onReorderButtonClicked(v);
+            }
+        });
+        mRotateButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(final View v) {
+                onRotateButtonClicked();
+            }
+        });
+        mDeleteButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(final View v) {
+                onDeleteButtonClicked();
+            }
+        });
+    }
+
+    private void onDeleteButtonClicked() {
+        final int deletedItem = mPreviewsPager.getCurrentItem();
```

Comment:
This method does a lot of things. It deletes locally, deletes remotely and updates the thumbnails collection view.